src/borg/repository.py

```diff
@@ -1500,20 +1517,21 @@ def recover_segment(self, segment, filename):
                 # memoryview context manager is problematic, see https://bugs.python.org/issue35686
                 data = memoryview(mm)
                 d = data
+                chunk = None
+                alignment_normal = True
                 try:
                     with open(filename, 'wb') as fd:
                         fd.write(MAGIC)
                         while len(d) >= self.header_fmt.size:
-                            crc, size, tag = self.header_fmt.unpack(d[:self.header_fmt.size])
```

Comment:
you see here: it never tries to unpack if there is not enough (>= self.header_fmt.size) data.